packages/shared/src/components/filters/MyFeedHeading.tsx

```diff
@@ -49,39 +51,45 @@ function MyFeedHeading({
   });
   const isRefreshing = feedQueryState?.status === 'success' && isFetchingFeed;
 
-  return (
-    <>
-      {forceRefresh && (
+  const RenderFeedActions = () => {
+    return (
+      <>
+        {forceRefresh && (
+          <Button
+            size={
+              shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium
+            }
+            variant={ButtonVariant.Float}
+            className="mr-auto"
+            onClick={onRefresh}
+            icon={<RefreshIcon />}
+            iconPosition={
+              shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
+            }
+            loading={isRefreshing}
+          >
+            {!isMobile && !isNewMobileLayout ? 'Refresh feed' : null}
+          </Button>
+        )}
         <Button
           size={
             shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium
           }
           variant={ButtonVariant.Float}
           className="mr-auto"
-          onClick={onRefresh}
-          icon={<RefreshIcon />}
+          onClick={onClick}
+          icon={<FilterIcon />}
           iconPosition={
             shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
           }
-          loading={isRefreshing}
         >
-          {!isMobile ? 'Refresh feed' : null}
+          {!isMobile && !isNewMobileLayout ? 'Feed settings' : null}
         </Button>
-      )}
-      <Button
-        size={shouldUseMobileFeedLayout ? ButtonSize.Small : ButtonSize.Medium}
-        variant={ButtonVariant.Float}
-        className="mr-auto"
-        onClick={onClick}
-        icon={<FilterIcon />}
-        iconPosition={
-          shouldUseMobileFeedLayout ? ButtonIconPosition.Right : undefined
-        }
-      >
-        {!isMobile ? 'Feed settings' : null}
-      </Button>
-    </>
-  );
+      </>
+    );
+  };
+
+  return <RenderFeedActions />;
```

Comment:
Why do we still have this wrapped in a component? I don't think it's needed.

Reply:
good catch, this was leftover from some intermediary changes